src/backend/base/langflow/components/models/embedding_model.py

```
@@ -77,37 +78,53 @@
+                max_retries=self.max_retries,
+                timeout=self.request_timeout or None,
+                show_progress_bar=self.show_progress_bar,
+                model_kwargs=self.model_kwargs or {},
             )
-        msg = f"Unknown provider: {provider}"
-        raise ValueError(msg)
+        if provider == "SentenceTransformers":
+            try:
+                from sentence_transformers import SentenceTransformer
+            except ImportError as e:
+                raise ImportError("Please install sentence-transformers to use this provider.") from e
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)